Adopt strict mode in phpcs/ (41 pre-existing null-safety findings)

`phpcs/tsconfig.json` now carries an explicit `"strict": false`. This tracks removing it.

## These are pre-existing, not a TypeScript 7 regression

Measured under both compilers with `--strict`:

| Project | TS 5.9.3 | TS 7.0.2 |
|---|---|---|
| `phpcs-server` | 0 | 0 |
| `phpcs` | **41** | **41** |

Identical counts. The debt has been there all along; TypeScript 7 only surfaced it by defaulting `strict` to true where TypeScript 5 defaulted it false. Setting `"strict": false` explicitly preserves exactly the semantics this project has always had — it suppresses nothing that was previously enforced.

`phpcs-server` passes strict under both, so it now declares `"strict": true` and keeps it.

## What the 41 are

| Count | Code | Meaning |
|---|---|---|
| 10 | `TS2322` | `boolean \| undefined` not assignable to `boolean` |
| 4 | `TS2322` | `string \| null \| undefined` not assignable to `string \| null` |
| 3 | `TS2454` | `folder` used before being assigned |
| 3 | `TS2322` | `number \| undefined` not assignable to `number` |
| 2 | `TS2345` | `string \| undefined` argument for a `string` parameter |
| 2 | `TS2322` | `string[] \| undefined` not assignable to `string[]` |
| 2 | `TS2322` | `null` not assignable to `string` |
| 3 | `TS2564` | property has no initializer (`statusBarItem`, `timer`, `standardStatusBarItem`) |
| … | `TS18048` | `process.env.PATH` possibly `undefined` |

Concentrated in `phpcs/src/configuration.ts`, `phpcs/src/status.ts` and `phpcs/src/resolvers/*.ts`.

Several look like genuine latent bugs rather than noise — `process.env.PATH` really can be undefined, and the resolvers returning `null` where a `string` is declared is the same class of mismatch that TypeScript 7 caught in `makeSettings` (fixed in the TS 7 upgrade).

## Suggested approach

Not one sweep. `strictPropertyInitialization` (the 3 `TS2564`) and `strictNullChecks` (most of the rest) can be enabled independently, so they can land as separate reviewable changes before the umbrella flag flips:

1. `strictPropertyInitialization` — 3 findings, mechanical
2. `strictNullChecks` — the remainder, needs real thought about which values are genuinely nullable
3. Replace `"strict": false` with `"strict": true`, matching `phpcs-server`

## Related

The client is also the half with no meaningful test coverage — one placeholder assertion, and nothing exercising the LSP handshake (noted when LSP 3.18 landed in #71). Worth bearing in mind: these fixes will not be caught by tests if they go wrong.

**Correcting the staging plan above — the order I suggested is impossible.**

`strictPropertyInitialization` cannot be enabled first:

```
error TS5052: Option 'strictPropertyInitialization' cannot be specified
without specifying option 'strictNullChecks'.
```

So the flags cannot be staged independently. Measured breakdown for `phpcs/` under 7.0.2:

| Flag | Errors |
|---|---|
| `strictNullChecks` | **35** |
| `strictNullChecks` + `strictPropertyInitialization` | **41** |
| `strictFunctionTypes` | 0 |
| `strictBindCallApply` | 0 |
| `noImplicitThis` | 0 |
| `useUnknownInCatchVariables` | 0 |
| `alwaysStrict` | 0 |

Only two sub-flags do anything, and one depends on the other. `noImplicitAny` is already on separately.

## Stage by area instead

The 35 `strictNullChecks` errors are concentrated, and split cleanly along a line that matters for testability:

| File | Errors | Needs the extension host? |
|---|---|---|
| `src/configuration.ts` | 28 | **yes** — imports `vscode` |
| `src/resolvers/composer-path-resolver.ts` | 4 | no |
| `src/resolvers/global-path-resolver.ts` | 2 | no |
| `src/resolvers/path-resolver.ts` | 1 | no |

The resolvers are **pure** — no `vscode` import — so they can be unit tested with tsx and mocha exactly as `phpcs-server` is, no extension host needed. That makes them the right place to start: 7 of the 41 sit in code we can put real tests around first.

`configuration.ts` and `status.ts` import `vscode` and need either extension-host tests or careful review.

## Revised plan

1. Unit tests for `src/resolvers/*` (tsx + mocha, mirroring the server suite), then fix their 7 findings with coverage proving the behaviour
2. Fix the 28 in `configuration.ts` — the largest and least covered piece
3. Flip `"strict": false` → `"strictNullChecks": true` once those are clean
4. Add `strictPropertyInitialization` and fix the remaining 6 (`status.ts`'s uninitialised `statusBarItem`, `timer`, `standardStatusBarItem`)
5. Collapse to `"strict": true`, matching `phpcs-server`

Steps 1–2 are the real work; 3–5 are mechanical once they are done.

--- a/phpcs/src/resolvers/composer-path-resolver.ts
+++ b/phpcs/src/resolvers/composer-path-resolver.ts
@@ -14,8 +14,12 @@ export class ComposerPhpcsPathResolver extends PhpcsPathResolverBase {
 	protected readonly _workspaceRoot: string;
 	protected readonly _workingPath: string;
 
-	protected _composerJsonPath: string;
-	protected _composerLockPath: string;
+	// Lazy caches, resolved on first access by the getters below. Left
+	// unresolved in the constructor because fs.realpathSync throws when the
+	// file is absent, which is a normal case here — hasComposerJson() exists
+	// precisely to ask.
+	protected _composerJsonPath: string | undefined;
+	protected _composerLockPath: string | undefined;
 
 	/**
 	 * Class constructor.

--- a/phpcs/src/status.ts
+++ b/phpcs/src/status.ts
@@ -17,15 +17,19 @@ import { Timer } from './timer';
 
 export class PhpcsStatus {
 
-	private statusBarItem: StatusBarItem;
+	// Created on first use by getStatusBarItem(), not in the constructor — a
+	// status bar item is only worth allocating once there is something to show.
+	private statusBarItem: StatusBarItem | undefined;
 	private documents: string[] = [];
 	private processing: number = 0;
 	private buffered: number = 0;
 	private spinnerIndex = 0;
 	private spinnerSequence: string[] = ["|", "/", "-", "\\"];
-	private timer: Timer;
+	// Likewise created on demand by getTimer().
+	private timer: Timer | undefined;
 	private channel: OutputChannel;
-	private standardStatusBarItem: StatusBarItem;
+	// Likewise created on demand by getStandardStatusBarItem().
+	private standardStatusBarItem: StatusBarItem | undefined;
 	private documentStandards: Map<string, string | null> = new Map();
 
 	public constructor()

--- a/phpcs/tsconfig.json
+++ b/phpcs/tsconfig.json
@@ -1,11 +1,10 @@
 {
 	"compilerOptions": {
-		// Not yet the umbrella "strict": true that phpcs-server carries — that
-		// also turns on strictPropertyInitialization, which still reports the
-		// uninitialised status bar fields in status.ts. strictNullChecks is on
-		// and clean; the rest of #78 is what closes the gap.
-		"strict": false,
-		"strictNullChecks": true,
+		// Matches phpcs-server. Reached in stages through #78: the resolvers, then
+		// configuration.ts, then the lazily-initialised fields this last flag
+		// covers. Explicit rather than inherited, because TypeScript 5 defaults
+		// it to false and 7 defaults it to true.
+		"strict": true,
 		"noUnusedLocals": true,
 		"noUnusedParameters": true,
 		"noImplicitAny": true,

